src/main/java/soolutions/hometaskapp/account/BasicAccount.java

```diff
@@ -22,12 +30,22 @@ public boolean open() {
     }
 
     @Override
-    public Amount balance() {
-      return amount;
+    public void apply(Transaction transaction) {
+        if (!transaction.amount().matchesCurrency(currency))
```

Comment:
I thinks we know too much about `Amount` object here.

Reply:
Any thought on how to change that?

Since Balance is now Lazy Evaluated, the previous unit test for adding EUR to a USD account wasn't throwing an exception. It makes business sense that one can't deposit EUR into a USD account.

Reply:
@SilasReinagel let's trust our `Amount` object to be "smart" for now. Policies will be implemented later (you should've left a puzzle)